File: src/main/java/com/daddyrusher/bankdepositsapp/domain/Deposit.java
package com.daddyrusher.bankdepositsapp.domain;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "deposits")
public class Deposit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "client_id")
    @NotBlank
    private Client client;

    @ManyToOne
    @JoinColumn(name = "bank_id")
    @NotBlank
    private Bank bank;

    @Column(name = "start_date")
    @NotBlank
    private LocalDateTime startDate;

    @Column(name = "percent")
    @NotBlank
    private Double percent;

    @Column(name = "period")
    @NotBlank
    private Integer period;
}

File: src/main/kotlin/com/daddyrusher/bankdeposits/domain/Deposit.kt
package com.daddyrusher.bankdeposits.domain;

import java.time.LocalDateTime
import javax.persistence.*;
import javax.validation.constraints.NotBlank;

@Entity
@Table(name = "deposits")
class Deposit : AbstractEntity() {
    @ManyToOne
    @JoinColumn(name = "client_id")
    @NotBlank
    private val client: Client = Client()

    @ManyToOne
    @JoinColumn(name = "bank_id")
    @NotBlank
    private val bank: Bank = Bank()

    @Column(name = "start_date")
    @NotBlank
    private val startDate: LocalDateTime? = null

    @Column(name = "percent")
    @NotBlank
    private val percent: Double = 0.0

    @Column(name = "period")
    @NotBlank
    private val period: Int = 0
}
